device/coordinate_manager.cpp

```diff
@@ -321,6 +322,75 @@ void CoordinateManager::fill_arc_logical_to_translated() {
     }
 }
 
+void CoordinateManager::fill_tensix_core_structures() {
+    std::vector<size_t> harvested_y_coords = get_harvested_indices(tensix_harvesting_mask);
+    for (size_t y = 0; y < tensix_grid_size.y; y++) {
+        for (size_t x = 0; x < tensix_grid_size.x; x++) {
+            const tt_xy_pair core = tensix_cores[y * tensix_grid_size.x + x];
+            CoreCoord core_coord(core.x, core.y, CoreType::TENSIX, CoordSystem::PHYSICAL);
+            if (std::find(harvested_y_coords.begin(), harvested_y_coords.end(), y) == harvested_y_coords.end()) {
+                unharvested_tensix_cores.push_back(core_coord);
+            } else {
+                harvested_tensix_cores.push_back(core_coord);
+            }
+        }
+    }
+    const size_t num_harvested_y = harvested_y_coords.size();
+    tensix_grid_size.y -= num_harvested_y;
+    harvested_tensix_grid_size.x = tensix_grid_size.x;
+    harvested_tensix_grid_size.y = num_harvested_y;
+}
+
+void CoordinateManager::fill_dram_core_structures() {
+    for (size_t bank = 0; bank < dram_grid_size.x; bank++) {
```

Comment:
You're using these two for loops only for accessing dram_cores. I don't see these indexes used for anything other than that (bank and port). Consider switching to 
```
for (tt_xy_pair& core: dram_core)
```

Also for all other fill_ functions. fill_tensix_core_structures can probably use core.y instead of y when searching harvested.

Reply:
I will change the loop, but it y and core.y are not equal, because y is logical and core.y is physical